web/src/AppController.ts

```diff
@@ -45,28 +45,32 @@ class AppController {
 
       let data = JSON.parse(event.data)
 
-      data.Resources = data.Resources.map((r: any) => {
-        if (r.ResourceInfo === null) {
-          r.ResourceInfo = {
-            PodName: "",
-            PodCreationTime: "",
-            PodUpdateStartTime: "",
-            PodStatus: "",
-            PodStatusMessage: "",
-            PodRestarts: 0,
-            PodLog: "",
-            YAML: "",
-            Endpoints: [],
-          }
-        }
-        r.Alerts = getResourceAlerts(r)
-        return r
-      })
+      data.Resources = this.setDefaultResourceInfo(data.Resources)
       // @ts-ignore
       this.component.setAppState({ View: data })
     })
   }
 
+  setDefaultResourceInfo(resources: Array<any>): Array<any> {
```

Comment:
i wonder how long it will be until we just switch over to js protobufs, or some other message format that automatically does the default population for us

Reply:
I would like to investigate solutions for this at some point because it is very painful. I've heard negative things about using protobufs for js but haven't looked in to it myself.